core/modules/ros_mdlw/kpsr_ros_core/src/ros_env.cpp

```diff
@@ -19,40 +19,84 @@
 
 #include "ros_env.h"
 #include <string.h>
+#include <stdexcept>
 
 kpsr::ros_mdlw::RosEnv::RosEnv(ros::NodeHandle * nodeHandle) {
 	this->nodeHandle = nodeHandle;
 }
 
-void kpsr::ros_mdlw::RosEnv::getPropertyString(const std::string key, std::string & value) {
-    nodeHandle->getParam(key, value);
+void kpsr::ros_mdlw::RosEnv::getPropertyString(const std::string& key, std::string & value, const std::string& rootNode) {
+    if (rootNode.size() == 0) {
```

Comment:
maybe we can create a small function or inline macro for this? I.e., calculate key for key + rootNode?

Reply:
Done